src/DataManagement/Saving/SavingCalculator.cs

```diff
@@ -4,113 +4,144 @@
-                        averagesOfAmountByImportanceDefault[i - 1] = sumsOfAmountByImportanceDefault[i - 1] / entryAmounts[i - 1];
-                    }
-                    else
+                    var listByMonth = listByImportance.GetBy(oldest);
+                    var monthTotal = 0M;
+
+                    foreach (var entry in listByMonth)
                     {
-                        averagesOfAmountByImportanceAdjusted[i - 1] = 0;
-                        averagesOfAmountByImportanceDefault[i - 1] = 0;
+                        var amountAfterSaving = SavingType switch
+                        {
+                            SavingType.Minimal => entry.Amount * ratio * MinimalSavingValue,
+                            SavingType.Regular => entry.Amount * ratio * RegularSavingValue,
+                            SavingType.Maximal => entry.Amount * ratio * MaximalSavingValue,
+                            _ => throw new ArgumentOutOfRangeException()
+                        };
+                        amountAfterSaving = amountAfterSaving > entry.Amount ? entry.Amount : amountAfterSaving;
+                        EntrySuggestions.Add(new SavingSuggestion(entry, amountAfterSaving));
+                        monthTotal += entry.Amount;
                     }
 
-                    approximateSavedAmount += averagesOfAmountByImportanceDefault[i - 1] - averagesOfAmountByImportanceAdjusted[i - 1];
+                    importanceAverage += monthTotal;//listByMonth.Average(x => x.Amount));
+                    months++;
+                    oldest = TimeManager.MoveToNextMonth(oldest);
                 }
-                timesToRepeatSaving++;
+
+                importanceAverage /= months > 0 ? months : 1;
+
+                var newAverage = SavingType switch
+                {
+                    SavingType.Minimal => importanceAverage * ratio * MinimalSavingValue,
+                    SavingType.Regular => importanceAverage * ratio * RegularSavingValue,
+                    SavingType.Maximal => importanceAverage * ratio * MaximalSavingValue,
+                    _ => throw new ArgumentOutOfRangeException()
+                };
+
+                SavingSuggestionByImportanceList.Add(new SavingSuggestionByImportance(newAverage, importanceAverage, (Importance)i));
             }
-            for (var i = enumCount; i > (int)Importance.Necessary; i--)
+        }
+
+        private void CalculateSavingTime()
+        {
+            var increasedMonthlySavings = SavingSuggestionByImportanceList.Sum(savingSuggestionByImportance => savingSuggestionByImportance.OldAverage - savingSuggestionByImportance.NewAverage);
+            var amountToSave = Goal.Amount - StartingBalance;
+            MonthsToSave = (int)Math.Ceiling(decimal.Divide(amountToSave, MonthlyIncome + increasedMonthlySavings));
+            if (MonthsToSave > 1000)
```

Comment:
I think 1000 months is overkill and max time to save should be reduced to, let's say 5 years? (60 months)

Reply:
I'd say something like 20 years would be at the upper part of reasonable, what do you think?

Reply:
Maybe something like if item is bellow 10k price - 20 years, bellow 5 years? But this creates more problems as different value in different currencies, so maybe middle ~12 years? :D

Reply:
I just set to 10 years for simplicity :D